compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -1473,6 +1478,11 @@ impl ReprOptions {
         let mut size = None;
         let mut max_align: Option<Align> = None;
         let mut min_pack: Option<Align> = None;
+
+        // Generate a deterministically-derived seed from the item's path hash
+        // to allow for incremental & cross-crate compilation to actually work
```

Comment:
Incremental would probably be fine. Different crates with different values of the `-Z` flag is probably more important.